auth0/index.d.ts

```diff
@@ -217,6 +217,58 @@ export class AuthenticationClient {
   getCredentialsGrant(scope: string): Promise<any>;
   getCredentialsGrant(scope: string, cb: (err: Error, message: string) => void): void;
 
+  database: {
+    signIn(userData: UserData): Promise<any>;
+    signIn(userData: UserData, cb: (err: Error, message: string) => void): void;
+
+    signUp(userData: UserData): Promise<any>;
+    signUp(userData: UserData, cb: (err: Error, message: string) => void): void;
+
+    changePassword(userData: UserData): Promise<any>;
+    changePassword(userData: UserData, cb: (err: Error, message: string) => void): void;
+
+    requestChangePasswordEmail(userData: UserData): Promise<any>;
+    requestChangePasswordEmail(userData: UserData, cb: (err: Error, message: string) => void): void;
+  };
+
+  oauth: {
+    socialSignIn(userData: { access_token: string; connection: string; }): Promise<any>;
+    socialSignIn(userData: { access_token: string; connection: string; }, cb: (err: Error, message: string) => void): void;
+
+    signIn(userData: UserData): Promise<any>;
+    signIn(userData: UserData, cb: (err: Error, message: string) => void): void;
+
+    getClientCredentialsGrant(userData: UserData): Promise<any>;
+    getClientCredentialsGrant(userData: UserData, cb: (err: Error, message: string) => void): void;
+  };
+  passwordless: {
+    signIn(userData: UserData): Promise<any>;
+    signIn(userData: UserData, cb: (err: Error, message: string) => void): void;
+
+    sendEmail(userData: { email: string; send: 'link' | 'code' }): Promise<any>;
+    sendEmail(userData: { email: string; send: 'link' | 'code' }, cb: (err: Error, message: string) => void): void;
+
+    sendSMS(userData: { phone_number: string; client_id?: string }): Promise<any>;
+    sendSMS(userData: { phone_number: string; client_id?: string }, cb: (err: Error, message: string) => void): void;
+
+
+  };
+  users: {
+
+    getInfo(userData: { accessToken: string }): Promise<any>;
```

Comment:
I know this wasn't done before, but can we add stricter return types than `Promise<any>`?